iroha_p2p/src/peer.rs

```diff
@@ -103,16 +112,20 @@ where
             None => (None, None),
             Some((read, write)) => (Some(read), Some(write)),
         };
+        let outgoing = read.is_none() && write.is_none();
         Ok(Self {
             id,
             connection_id,
+            peer_connection_id: 0,
             read,
             write,
             state,
+            outgoing,
             secret_key,
             public_key,
             cipher: None,
             broker,
+            finish_sender: None,
             _key_exchange: PhantomData::default(),
```

Comment:
14 fields is kinda too much to me. Like it needs to be squashed and some flag fields needs to be removed

Reply:
The types of that fields are very different. The only way is to create another struct that will hold some of the fields from here. But it will add complexity.

Reply:
It is already too complex to my mind. Like more than it needs to